apps/server/src/form-instances/form-instances.service.ts

```diff
@@ -300,6 +300,42 @@ export class FormInstancesService {
       ...updatedSignature,
     };
 
-    return formInstance;
+    const allSigned = formInstance.signatures.every((sig) => sig.signed);
+
+    if (allSigned) {
+      await this.prisma.formInstance.update({
+        where: { id: formInstanceId },
+        data: { completed: true, completedAt: new Date() },
+      });
+    }
+
+    const updatedFormInstance = await this.findOne(formInstanceId);
+
+    if (!updatedFormInstance) {
+      throw new NotFoundException(
+        FormInstanceErrorMessage.FORM_INSTANCE_NOT_FOUND,
+      );
+    }
+
+    return updatedFormInstance;
+  }
+
+  async markFormInstanceAsCompleted(formInstanceId: string) {
+    const formInstance = await this.prisma.formInstance.findUnique({
+      where: { id: formInstanceId },
+    });
+
+    if (!formInstance) {
```

Comment:
we can also consider adding handling in cases where the form instance is already marked completed, and when the form instance is not ready to be marked as completed